<launch>
  <rosparam>
    base/names: [ "Wheel1", "Wheel2", "Wheel3" ]
    base/families: [ "Rosie" ]

    arm/names: [ "Base", "Shoulder", "Elbow", "Wrist1", "Wrist2", "Wrist3" ]
    arm/families: [ "Rosie" ]
    arm/gains_package: "hebi_cpp_api_examples"
    arm/gains_file: "config/gains/a-2085-06_gains.xml"
    arm/hrdf_package: "hebi_cpp_api_examples"
    arm/hrdf_file: "config/hrdf/a-2085-06.hrdf"
    arm/home_position: [ 0, 1.69517, 2.16275, 2.44159, -1.39, 1.57459 ]
    arm/enabled_paths: [ ]
    arm/enabled_waypoints: [ ]

    gripper/name: "Spool"
    gripper/family: "Rosie"
    gripper/gains_package: "hebi_cpp_api_examples"
    gripper/gains_file: "config/gains/gripper_spool_gains.xml"
  </rosparam>

  <!-- Robot components -->
  <node ns="base" pkg="hebi_cpp_api_examples" type="omni_base_node" name="omni_base_node" output="screen" required="true"/>
  <node ns="arm" pkg="hebi_cpp_api_examples" type="teach_repeat_node" name="teach_repeat_node" output="screen" required="true"/>
  <node ns="gripper" pkg="hebi_cpp_api_examples" type="gripper_node" name="gripper_node" output="screen" required="true"/>

  <!-- Mobile IO controller -->
  <node pkg="hebi_cpp_api_examples" type="rosie_teach_repeat_controller_node" name="rosie_teach_repeat_controller_node" output="screen" required="true">
    <remap from="/cmd_vel" to="/base/cmd_vel"/>

    <remap from="/playback" to="/arm/playback"/>
    <remap from="/compliant_mode" to="/arm/compliant_mode"/>
    <remap from="/save_waypoint" to="/arm/save_waypoint"/>

    <remap from="/gripper_strength" to="/gripper/gripper_strength"/>
  </node>

</launch>
